packages/grid_client/src/modules/capacity.ts

```diff
@@ -43,13 +46,26 @@ class Capacity {
   @expose
   @validateInput
   async filterNodes(options?: FilterOptions): Promise<NodeInfo[]> {
-    if (options.farmName) {
+    const twinId = this.config.twinId;
+    if (options?.farmName) {
       options.farmId = await this.nodes.getFarmIdFromFarmName(options.farmName);
     }
-    if (options.farmId) {
+
+    if (options?.farmId && options.rentedBy === twinId) {
+      const proxyURL = this.config.proxyURL;
+      return send("get", urlJoin(proxyURL, `/nodes?rented_by=${twinId}`), "", {})
+        .then(res => {
+          return res;
+        })
+        .catch(err => {
+          throw Error(`Error while getting nodes: ${err}`);
+        });
+    }
+
+    if (options?.farmId) {
```

Comment:
in case the `rentedBy` is set, we shouldn't use the farmer bot and just fall back to the gridporxy
```suggestion
    if (options?.farmId && !options.rentedBy) {
```